anvil/src/eth/api.rs

```diff
@@ -1476,6 +1477,49 @@ impl EthApi {
         self.backend.load_state(buf).await
     }
 
+    /// Retrieves the Anvil node configuration params.
+    ///
+    /// Handler for RPC call: `anvil_nodeInfo`
+    pub async fn anvil_node_info(&self) -> Result<NodeInfo> {
+        node_info!("anvil_nodeInfo");
+
+        let env = self.backend.env().read();
+        let fork_config = self.backend.get_fork();
+        let tx_order = self.transaction_order.read();
+
+        Ok(NodeInfo {
+            current_block_number: self.backend.best_number(),
+            current_block_timestamp: env.block.timestamp.try_into().unwrap_or(u64::MAX),
+            current_block_hash: self.backend.best_hash(),
+            hard_fork: env.cfg.spec_id,
+            transaction_order: match *tx_order {
+                TransactionOrder::Fifo => "fifo".to_owned(),
+                TransactionOrder::Fees => "fees".to_owned(),
+            },
+            environment: NodeEnvironment {
+                base_fee: self.backend.base_fee(),
+                chain_id: self.backend.chain_id(),
+                gas_limit: self.backend.gas_limit(),
+                gas_price: self.backend.gas_price(),
+            },
+            fork_config: match fork_config {
+                Some(fork) => {
+                    let config = fork.config.read();
+                    NodeForkConfig {
+                        fork_url: Some(config.eth_rpc_url.clone()),
+                        fork_block_number: Some(config.block_number),
+                        fork_retry_backoff: Some(config.backoff.as_millis()),
+                    }
+                }
+                None => NodeForkConfig {
+                    fork_url: None,
+                    fork_block_number: None,
+                    fork_retry_backoff: None,
+                },
+            },
```

Comment:
this can be simplified with `map(|fork|NodeForkCOnfig{..}).unwrap_or_default()`

Reply:
Done in https://github.com/foundry-rs/foundry/pull/3779/commits/79a8d669cdc3cdcedb3cfd6c5a7cc210e27ef884